reference/docs-conceptual/learn/shell/running-commands.md

```diff
@@ -0,0 +1,163 @@
+---
+description: >
+  This article shows how to run commands in PowerShell.
+title: Running commands in the shell
+ms.date: 08/22/2022
+---
+# Running commands in the shell
+
+PowerShell is a command-line shell and a scripting language used for automation. Similar to other
+shells, like `bash` on Linux or the Windows Command Shell (`cmd.exe`), PowerShell lets you to run
+any command available on your system, not just PowerShell commands.
+
+## Types of commands
+
+For any shell in any operating system there are three types of commands:
+
+- **Shell language keywords** are part of the shell's scripting language.
+
+  - Examples of `bash` keywords include: `if`, `then`, `else`, `elif`, and `fi`.
+  - Examples of `cmd.exe` keywords include: `dir`, `copy`, `move`, `if`, and `echo`.
+  - Examples of PowerShell keywords include: `for`, `foreach`, `try`, `catch`, and `trap`.
+
+  Shell language keywords can only be used within the runtime environment of the shell. There is no
+  executable file, external to the shell, that provides the keyword's functionality.
+
+- **OS-native commands** are executable files installed in the operating system. The executables can
+  be run from any command-line shell, like PowerShell. This includes script files that may require
+  other shells to work properly. For example, if you run a Windows batch script (`.cmd` file) in
+  PowerShell, PowerShell runs `cmd.exe` and passes in the batch file for execution.
+
+- **Shell environment-specifc commands** are commands defined in external files that can only be
+  used within the runtime environment of the shell. These include scripts and functions, or they can
+  be specially compiled modules that add commands to the shell runtime. In PowerShell, these command
+  are known as _cmdlets_ (pronounced "command-lets").
+
+## Running OS-native commands
+
+Any OS-native command can be run from the PowerShell command line. In most cases you run the command
+exactly has you would in `bash` or `cmd.exe`. The following example shows running the `grep` command
+in `bash` on Ubuntu Linux.
+
+```bash
+sdwheeler@circumflex:~$ grep sdwheeler /etc/passwd
+sdwheeler:x:1000:1000:,,,:/home/sdwheeler:/bin/bash
+sdwheeler@circumflex:~$ pwsh
+PowerShell 7.2.6
+Copyright (c) Microsoft Corporation.
+
+https://aka.ms/powershell
+Type 'help' to get help.
```

Comment:
Should this be here or in the next example block? It's referred to in the prose after this block but at first I went back to the preceding prose to see why it was being run.

Reply:
pwsh is run in bash. The output make more sense here.